src/saveload/compat/settings_sl_compat.h

```diff
@@ -0,0 +1,269 @@
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file settings_sl_compat.cpp Loading of savegames before table headers were added. */
+
+#include "../../stdafx.h"
+
+#include "../saveload.h"
+
+#include "../../safeguards.h"
+
+/** The order of game options in the savegame prior to SLV_SETTINGS_BY_NAME. */
+const SaveLoadCompat _gameopt_sl_compat[] = {
+	SLC_VAR("diff_custom"),
+	SLC_VAR("diff_level"),
+	SLC_VAR("locale.currency"),
+	SLC_VAR("units"),
+	SLC_VAR("game_creation.town_name"),
+	SLC_VAR("game_creation.landscape"),
+	SLC_VAR("game_creation.snow_line_height"),
+	SLC_NULL(1, SLV_22, SLV_165),
+	SLC_NULL(1, SL_MIN_VERSION, SLV_23),
+	SLC_VAR("gui.autosave"),
```

Comment:
"gui.autosave" is SLF_NOT_IN_SAVE. So you can remove it from here?

Reply:
No, it is used between version 0 ... 23, so for loading the system still needs to know for those versions there are bytes there :)